mood logic

Nwanekwup opened the pull request:
- added more test questions
- created the logic  behind determining the predominant mood from personality test answers
- added a previous button for the test screen.

[mood logic prisma demo](https://www.loom.com/share/5e5b208acadb46c290012bebbb15187c?sid=32baa696-a1af-4fe9-ade2-704cb0d472c1)

Nwanekwup committed:
mood logic

- added more test questions
- created the logic  behind determining the predominant mood from personality test answers
- added a previous button for the test screen.

ChibuezeEjiofor commented on frontend/src/components/TakeQuiz.jsx:
```diff
@@ -24,8 +24,116 @@ const questions = [
+  };
+
+  // Map the questions to moods and update the scores
+  answers.forEach((answer, index) => {
```
Why do you need both the emotion and it's opposite if the value is essentially the inverse of it?

ChibuezeEjiofor commented on frontend/src/components/TakeQuiz.jsx:
```diff
@@ -24,8 +24,116 @@ const questions = [
     question: "I feel goofy most days",
     answers: [5, 4, 3, 2, 1],
   },
+  {
```
So right now, the questions, answers, and moods are all strongly coupled. What you could do is have each question have a mood field. That way, you could iterate over the answers and know which mood to adjust. This would make your code much more modular, and make it much easier to have multiple questions for a single mood, or potentially even allow users to create questions/quizes.

Nwanekwup committed:
Answer options debugging

- Debugged code to make database record the latest answer to a question if user options more than once without having duplicate answers of the same question in the database.
- Debugged code where highlighted answer to previous question is cleared  in next answer options in next question.

[Answer options debugging demo](https://www.loom.com/share/12eb2f1a39a34958ba89f0c90c3cf781?sid=3deb5faf-2bd6-4c6d-b764-539d6e0fa7f8)

Nwanekwup committed:
Added "mood" field to questions variable

I Added "mood" field to questions variable and adjusted related aspects to make the code more efficient.
